File: solutions/102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function levelOrder(root: TreeNode | null): number[][] {

    const res: number[][] = [];

    if(root === null) {
        return res;
    }

    let queue: TreeNode[] = [root];

    while(queue.length > 0) {

        const nextQueue: TreeNode[] = [];
        const currLevelVals: number[] = [];

        for(let i = 0; i < queue.length; i += 1) {

            const currNode = queue[i]

            currLevelVals.push(currNode.val);

            if(currNode.left !== null) {
                nextQueue.push(currNode.left);
            }

            if(currNode.right !== null) {
                nextQueue.push(currNode.right);
            }

        }

        res.push(currLevelVals);
        queue = nextQueue;

    }

    return res;
    
};

File: solutions/102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        
        result = []

        if root is None:
            return result

        my_stack = [[root]]

        while len(my_stack) > 0:
            curr_level_vals = []
            curr_level_nodes = my_stack.pop()
            next_level_nodes = []
            for node in curr_level_nodes:
                curr_level_vals.append(node.val)
                if node.left is not None:
                    next_level_nodes.append(node.left)
                if node.right is not None:
                    next_level_nodes.append(node.right)
            if len(next_level_nodes) > 0:
                my_stack.append(next_level_nodes)
            result.append(curr_level_vals)

        return result